### PR TITLE
Typing (api/os)

### PR DESCRIPTION
**Pull request by mq1**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding Issue #___
- [x] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: tooling/api/src/os.ts [hunk 1]
```diff
@@ -58,23 +61,29 @@ async function version(): Promise<string> {
   })
 }
 
+
+type Type = 'Linux' | 'Darwin' | 'Windows_NT'
```
Comment (amrbashir) [inline tooling/api/src/os.ts]: I think `OsType` might be a better name here.

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for your work, can you also export these types ? 
```ts
export type {
	Platform,
	Arch
	OsType
}
```

**Comment by mq1** [comment]
done!

File: tooling/api/src/os.ts [hunk 2]
```diff
@@ -58,23 +61,29 @@ async function version(): Promise<string> {
...
+
+type OsType = 'Linux' | 'Darwin' | 'Windows_NT'
+
 /**
  * Returns `'Linux'` on Linux, `'Darwin'` on macOS, and `'Windows_NT'` on Windows.
  */
-async function type(): Promise<string> {
-  return invokeTauriCommand<string>({
+async function type(): Promise<Type> {
+  return invokeTauriCommand<Type>({
```
Comment (amrbashir) [inline tooling/api/src/os.ts]:
```suggestion
async function type(): Promise<OsType> {
  return invokeTauriCommand<OsType>({
```

**Review by amrbashir — APPROVED** [review]

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**